src/EditorFeatures/Test/CommentSelection/ToggleLineCommentCommandHandlerTests.cs

```diff
@@ -0,0 +1,1084 @@
+    {
+        [WpfFact, Trait(Traits.Feature, Traits.Features.ToggleLineComment)]
+        public void AddComment_EmptyCaret()
+        {
+            var markup = @"$$";
+            var expected = @"[||]";
+
+            ToggleComment(markup, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.ToggleLineComment)]
+        public void AddComment_EmptySelection()
+        {
+            var markup = @"[| |]";
+            var expected = @"[||]";
+
+            ToggleComment(markup, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.ToggleLineComment)]
+        public void AddComment_CaretInUncommentedLine()
+        {
+            var markup =
+@"
+class C
+{
+    void M()
+    {
+        var$$ i = 1;
+    }
+}";
+            var expected =
+@"
+class C
+{
+    void M()
+    {
+[|        //var i = 1;|]
+    }
+}";
+
+            ToggleComment(markup, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.ToggleLineComment)]
+        public void AddComment_CaretBeforeUncommentedLine()
+        {
+            var markup =
+@"
+class C
+{
+    void M()
+    {
+$$        var i = 1;
+    }
+}";
+            var expected =
+@"
+class C
+{
+    void M()
+    {
+[|        //var i = 1;|]
+    }
+}";
+
+            ToggleComment(markup, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.ToggleLineComment)]
+        public void AddComment_SingleLineSelected()
+        {
+            var markup =
+@"
+class C
+{
+    void M()
+    {
+        [|var i = 1;|]
+    }
+}";
+            var expected =
+@"
+class C
+{
+    void M()
+    {
+[|        //var i = 1;|]
+    }
+}";
+
+            ToggleComment(markup, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.ToggleLineComment)]
+        public void AddComment_PartialSingleLineSelected()
+        {
+            var markup =
+@"
+class C
+{
+    void M()
+    {
+        var [|i = 1;|]
+    }
+}";
+            var expected =
+@"
+class C
+{
+    void M()
+    {
+[|        //var i = 1;|]
+    }
+}";
+
+            ToggleComment(markup, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.ToggleLineComment)]
+        public void AddComment_SingleLineWithWhitespaceSelected()
+        {
+            var markup =
+@"
+class C
+{
+    void M()
+    {
+[|
+        var i = 1;
+   |]
+    }
+}";
+            var expected =
+@"
+class C
+{
+    void M()
+    {
+[|
+        //var i = 1;
+   |]
+    }
+}";
+
+            ToggleComment(markup, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.ToggleLineComment)]
+        public void AddComment_SelectionInsideCommentAtEndOfLine()
+        {
+            var markup =
+@"
+class C
+{
+    void M()
+    {
+        var i = 1; // A [|comment|].
+    }
+}";
+            var expected =
+@"
+class C
+{
+    void M()
+    {
+[|        //var i = 1; // A comment.|]
```

Comment:
is putting // right before first token same behavior as existing line comment feature? rather than adding a space after // ?

Reply:
Yep!